server/src/main/java/org/eclipse/lsp/cobol/service/DisposableLSPStateService.java

```diff
@@ -28,4 +30,15 @@
    * @return int
    */
   int getExitCode();
+
+  /** Signals the server to shutdown by updating the server state to SHUTDOWN_EXIT_CODE */
+  void shutdown();
+
+  /** Revoke shutdown for testing purpose. */
+  @VisibleForTesting
+  void revokeShutdown();
+
+  default boolean isServerShutdown() {
+    return this.getExitCode() == DisposableLSPStateService.SHUTDOWN_EXIT_CODE;
+  }
```

Comment:
This can confuse. In theory, an exit code can be any number. The exit code `0` means that the program ends correctly. `1` is a valid exit code, but this means that the program failed.
From that point, this checking does not look correct. Strictly saying the function returns true if the service shuts down without an error. Maybe it will be better to use a boolean to keep the service state?

Reply:
We use these exit codes in cobolLanguageServer. When the client sends an exit notification to the server (exit method).
So, if we receive an exit notification without any previous shutdown, we exit with 1 ( not ideal). If the order of notification is, 1st shutdown and then exit, then we exit with code 0.
So, do you suggest always exit with code 0 in the exit method and make the server state boolean?

Reply:
I see how it used and we did it the same way before. But this new function is not feeling right.
The method works and I'm OK to merge it.
My idea is to have the logic like the following:
`DisposableLSPStateService` works with boolean and keeps tracking UP state.
In `CobolLanguageServer::exit` we analyse this UP state. If the service state is down - everything is good and we can exit with `0`. If we see that the service is still running we can put a warning message into the log and exit with `1`.

It is not a significant change, but it requires rewriting unit tests and other places. Looks like I want to overcomplicate this.